src/query.js

```diff
@@ -77,6 +81,15 @@ class Query {
     return result;
   }
 
+  find(fn) {
+    let iterator = this.generator();
+    for (let next = iterator.next(); !next.done; next = iterator.next()) {
```

Comment:
I think we can even make it simpler if we take into account that the query itself is iterable:

```js
for (let item of this) {
  if (fn(item) {
    return item;
  }
}
```

Reply:
Can the same be done with reduce? (separate PR)

Reply:
Nice catch! It could totally.